### PR TITLE
Add possibility to close channel

### PR DESCRIPTION
**Pull request by maraph**
## Description
First PR for feature #21

## Motivation and Context
User has now the ability to close a channel within the app.

## How Has This Been Tested?
Tested on Simnet with Pixel 3/API 29 

## Screenshots (if appropriate):
close_button

close_dialog

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [Contribution document](../docs/CONTRIBUTING.md).
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by michaelWuensch** [comment]
Thanks for the PR.
I think the confirm dialog should show the Node Alias, not the public key. That would look much nicer.

**Comment by michaelWuensch** [comment]
I think we need to give the user the information how long a force close will take. I would like to have this information as readable time, not in blocks. Probably we should show this in two places:
1. On the confirm dialog. (Adding something like:  It will take 2 days 5 hours until your funds are available again)
2. On the channel details of a pending force close channel.  (A closing countdown, like the expiry contdown on active invoices)

**Comment by michaelWuensch** [comment]
One further thing we should change for closing channels is that in the transaction history the according On-Chain received transactions are linked (if possible) to the channel event and therefore displayed as channel event whithout showing a positive amount.
This is nothing that was introduced in this PR, but something that is related to the topic.
If we want to keep PRs small, we can also do this in a separate PR.

**Comment by maraph** [comment]
Thanks for the input. 
I've added an issue for the delays: https://github.com/LN-Zap/zap-android/issues/94
Regarding the second issue lets discuss in slack 👍 

I want to keep this PR as small as possible ;)

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -131,7 +131,12 @@
     <string name="channel_state_pending_force_closing">pending force close</string>
     <string name="channel_state_waiting_close">waiting close</string>
     <string name="no_channels">No channels</string>
-
+    <string name="channel_closing_transaction">Closing Transaction</string>
+    <string name="channel_close">Close Channel</string>
+    <string name="channel_close_force">Force Close Channel</string>
+    <string name="channel_close_error">Error occurred while closing channel</string>
+    <string name="channel_close_confirmation">Do you really want to close the channel with:\n%s</string>
```
Comment (michaelWuensch) [inline app/src/main/res/values/strings.xml]: I would actually add a \n at the end as well. Makes the message more readable in my opinion

File: app/src/main/res/layout/bsd_channeldetail.xml [hunk 2]
```diff
@@ -228,4 +297,43 @@
...
+
+    <Button
+        android:id="@+id/channelCloseButton"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="8dp"
+        android:layout_marginEnd="8dp"
+        android:background="@drawable/bg_clickable_item"
+        android:textColor="@color/lightningOrange"
+        android:visibility="gone"
```
Comment (michaelWuensch) [inline app/src/main/res/layout/bsd_channeldetail.xml]: Maybe add android:textAllCaps="false"
We do not have AllCaps buttons anywhere else (except the setup wallet button which has a completely different style anyway)

**Pull request merged**